src/with-render.js

```diff
@@ -1,18 +1,18 @@
 import { HTMLElement } from './util';
 
 export const withRender = (Base = HTMLElement) => class extends Base {
-  // Called to render the component.
-  renderCallback () {}
-
-  // Called after the component has rendered.
-  renderedCallback () {}
-
-  // Called to render the component.
-  rendererCallback (render) {
+  propsUpdatedCallback (next, prev) {
+    super.propsUpdatedCallback(next, prev);
     if (!this.shadowRoot) {
       this.attachShadow({ mode: 'open' });
     }
-    render(this.renderCallback(this), this.shadowRoot);
+    this.rendererCallback(this.shadowRoot, () => this.renderCallback(this));
```

Comment:
A possible issue here is creating an extra function every time we render. I'm not convinced there's actually an issue here. Incremental DOM uses this method in its examples.